cmd/github-ldap-user-group-creator/README.md

```diff
@@ -1,18 +1,29 @@
 # github-ldap-user-group-creator
 
-Given the mapping from github login to Red Had LDAP kerberos ID, this tool maintains a user group
-for each user in the map on every cluster in our build farm.
-
-For example, `m(some_name_at_github)=another_name_at_ldap` leads to a group
-
-```yaml
-apiVersion: user.openshift.io/v1
-kind: Group
-metadata:
-  name: some_name_at_github-group
-users:
-  - some_name_at_github
-  - another_name_at_ldap
-```
-
-The group `some_name_at_github-group` will be used by `ci-operator` which promotes the group to the admins of the namespace created for the test.
+## What it does
+
+`github-ldap-user-group-creator` is a tool to maintain the groups on CI clusters.
+
+
+## Why it exists
+
+The groups are used
+- by `ci-operator` which promotes the group to the admins of the namespace created for the test.
+- in [the manifests](https://github.com/openshift/release/tree/master/clusters) of CI clusters
+  in the release repo.
+
+## How it works
+
+`github-ldap-user-group-creator` reads 
+- the mapping files generated by [another tool](../ldap-users-from-github-owners-files) `ldap-users-from-github-owners-files`
+that stores the mapping from `github-id` to its Red Hat `kerberos-id` and for each `github-id`, creates a group `github-id-group`
+on each cluster.
+
+- the groups file generated by [another tool](../syn-rover-groups) that stores the group names and their members from 
+the Red Hat LDAP server and for each group creates a group on each cluster.
+
+## How is it deployed
+
+The periodic
+job [periodic-github-ldap-user-group-creator](https://deck-internal-ci.apps.ci.l2s4.p1.openshiftapps.com/?job=periodic-github-ldap-user-group-creator) ([definition](https://github.com/openshift/release/blob/d5ac4fc821a8c366a02ce47cd47e3d774c98497c/ci-operator/jobs/infra-periodics.yaml#L2157-L2199))
```

Comment:
I don't know that there is much value in adding the exact lines of the job definition. That seems likely to change a ton and get immediately out of date. Maybe just a link to the `infra-periodics.yaml` file saying it lives in there?

Reply:
It is a permalink which is not going to be out of date.
https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files
The content of the job could be tough.

The link to infra job file could be changed as well (i know it is more unlikely).
I do not have a strong opinion on either of them: just we have got many of those permalinks already. :)

Reply:
Updated.